modules/nf-core/minimap2/align/tests/main.nf.test

```diff
@@ -137,14 +140,113 @@ nextflow_process {
                     process.out.versions
                 ).match() }
             )
         }
 
     }
 
+    test("sarscov2 - bam, fasta, true, [], false, false") {
+
+        when {
+            process {
+                """
+                input[0] = [
+                    [ id:'test', single_end:true ], // meta map
+                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test3.single_end.markduplicates.sorted.bam', checkIfExists: true)
+                ]
+                input[1] = [
+                    [ id:'test_ref' ], // meta map
+                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)
+                ]
+                input[2] = true
+                input[3] = []
+                input[4] = false
+                input[5] = false
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(
+                    bam(process.out.bam[0][1]).getHeader(),
+                    bam(process.out.bam[0][1]).getReadsMD5(),
+                    process.out.versions
+                ).match() }
+            )
+        }
+
+    }
+
+    test("sarscov2 - bam, fasta, true, 'bai', false, false") {
+
+        when {
+            process {
+                """
+                input[0] = [
+                    [ id:'test', single_end:true ], // meta map
+                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test3.single_end.markduplicates.sorted.bam', checkIfExists: true)
+                ]
+                input[1] = [
+                    [ id:'test_ref' ], // meta map
+                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)
+                ]
+                input[2] = true
+                input[3] = 'bai'
+                input[4] = false
+                input[5] = false
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(
+                    bam(process.out.bam[0][1]).getHeader(),
+                    bam(process.out.bam[0][1]).getReadsMD5(),
+                    file(process.out.index[0][1]).name,
+                    process.out.versions
+                ).match() }
+            )
+        }
+
+    }
+
+    test("sarscov2 - bam, [], true, false, false") {
+
+        when {
+            process {
+                """
+                input[0] = [
+                    [ id:'test', single_end:true ], // meta map
+                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test3.single_end.markduplicates.sorted.bam', checkIfExists: true)
+                ]
+                input[1] = [
+                    [ id:'test_ref' ], // meta map
+                    []
```

Comment:
Are you testing if a reference is empty here, right?

Reply:
Yes, which will not work because the module then assigns the reads as reference, and our reads are bam-files.

The process in this test should fail with `error("BAM input requires reference")`, see https://github.com/nf-core/modules/actions/runs/10093279298/job/27914770643?pr=6021#step:14:154

Reply:
Nice!

Reply:
Sorry, that was the stub. Here's the right link https://github.com/nf-core/modules/actions/runs/10093279298/job/27914770643?pr=6021#step:14:83

Reply:
No worries, I understood what you meant.